### ISSUE TITLE
Wrong indentation on C# lambda

### ISSUE THREAD

**Issue by mihaipopescu** [uncrustify/uncrustify#1269]
```diff
@@ -13,20 +13,20 @@
         private static Type[] GetAllVisualElementTypes()
         {
             return typeof(VisualElement).Assembly.GetTypes()
-                .Where(t => t != typeof(VisualElement) &&
-                            t != typeof(Panel) &&
-                            !t.IsAbstract &&
-                            !typeof(IMElement).IsAssignableFrom(t) &&
-                            !typeof(IMContainer).IsAssignableFrom(t) &&
-                            typeof(VisualElement).IsAssignableFrom(t)).ToArray();
+                .Where(t => t != typeof(VisualElement) &&
+                t != typeof(Panel) &&
+                !t.IsAbstract &&
+                !typeof(IMElement).IsAssignableFrom(t) &&
+                !typeof(IMContainer).IsAssignableFrom(t) &&
+                typeof(VisualElement).IsAssignableFrom(t)).ToArray();
         }

         private static Type[] GetAllVisualElementAssetTypes()
         {
             return typeof(VisualElementAsset).Assembly.GetTypes()
                 .Where(t => t != typeof(VisualElementAsset) &&
-                            !t.IsAbstract &&
-                            typeof(VisualElementAsset).IsAssignableFrom(t)).ToArray();
+                !t.IsAbstract &&
+                typeof(VisualElementAsset).IsAssignableFrom(t)).ToArray();
         }
     }
 }
```
maybe we can have something like:
```
            return typeof(VisualElement).Assembly.GetTypes()
                .Where(t => t != typeof(VisualElement) &&
                            t != typeof(Panel) &&
                            !t.IsAbstract &&
                            !typeof(IMElement).IsAssignableFrom(t) &&
                            !typeof(IMContainer).IsAssignableFrom(t) &&
                            typeof(VisualElement).IsAssignableFrom(t)).ToArray();
```
or at least
```
  return typeof(VisualElement).Assembly.GetTypes()
                .Where(t => t != typeof(VisualElement) &&
                    t != typeof(Panel) &&
                    !t.IsAbstract &&
                    !typeof(IMElement).IsAssignableFrom(t) &&
                    !typeof(IMContainer).IsAssignableFrom(t) &&
                    typeof(VisualElement).IsAssignableFrom(t)).ToArray();
```

Expected versus output diff:
---
```diff
diff --git "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\60023-UNI-18437.cs" "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\60023-UNI-18437.cs"
index 63c64c1..4403c7b 100644
--- "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\60023-UNI-18437.cs"
+++ "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\60023-UNI-18437.cs"
@@ -1,42 +1,42 @@
 private static Type[] GetAllVisualElementTypes()
 {
     return typeof(VisualElement).Assembly.GetTypes()
-        .Where(t => t != typeof(VisualElement) &&
-                t != typeof(Panel) &&
-                !t.IsAbstract &&
-                !typeof(IMElement).IsAssignableFrom(t) &&
-                !typeof(IMContainer).IsAssignableFrom(t) &&
-                typeof(VisualElement).IsAssignableFrom(t)).ToArray();
+           .Where(t => t != typeof(VisualElement) &&
+        t != typeof(Panel) &&
+        !t.IsAbstract &&
+        !typeof(IMElement).IsAssignableFrom(t) &&
+        !typeof(IMContainer).IsAssignableFrom(t) &&
+        typeof(VisualElement).IsAssignableFrom(t)).ToArray();
 }
 
 private static Type[] GetAllVisualElementAssetTypes()
 {
     return typeof(VisualElementAsset).Assembly.GetTypes()
-        .Where(t => t != typeof(VisualElementAsset) &&
-                !t.IsAbstract &&
-                typeof(VisualElementAsset).IsAssignableFrom(t)).ToArray();
+           .Where(t => t != typeof(VisualElementAsset) &&
+        !t.IsAbstract &&
+        typeof(VisualElementAsset).IsAssignableFrom(t)).ToArray();
 }
 
 //We would like to get something like
 private static Type[] GetAllVisualElementAssetTypes()
 {
     return typeof(VisualElement).Assembly.GetTypes()
-        .Where(t => t != typeof(VisualElement) &&
-                    t != typeof(Panel) &&
-                    !t.IsAbstract &&
-                    !typeof(IMElement).IsAssignableFrom(t) &&
-                    !typeof(IMContainer).IsAssignableFrom(t) &&
-                    typeof(VisualElement).IsAssignableFrom(t)).ToArray();
+           .Where(t => t != typeof(VisualElement) &&
+        t != typeof(Panel) &&
+        !t.IsAbstract &&
+        !typeof(IMElement).IsAssignableFrom(t) &&
+        !typeof(IMContainer).IsAssignableFrom(t) &&
+        typeof(VisualElement).IsAssignableFrom(t)).ToArray();
 }
 
 // Or at least this
 private static Type[] GetAllVisualElementAssetTypes()
 {
     return typeof(VisualElement).Assembly.GetTypes()
-        .Where(t => t != typeof(VisualElement) &&
-            t != typeof(Panel) &&
-            !t.IsAbstract &&
-            !typeof(IMElement).IsAssignableFrom(t) &&
-            !typeof(IMContainer).IsAssignableFrom(t) &&
-            typeof(VisualElement).IsAssignableFrom(t)).ToArray();
+           .Where(t => t != typeof(VisualElement) &&
+        t != typeof(Panel) &&
+        !t.IsAbstract &&
+        !typeof(IMElement).IsAssignableFrom(t) &&
+        !typeof(IMContainer).IsAssignableFrom(t) &&
+        typeof(VisualElement).IsAssignableFrom(t)).ToArray();
 }
```

Debug support files:
---
[60023-UNI-18437.cs](https://gist.githubusercontent.com/mihaipopescu/756670a3de23ac714b3bd71dcc04e2e7/raw/63c64c10211459aa06e047e6d5d5cf70b8ae808f/60023-UNI-18437.cs)
[60023-UNI-18437.cs.log](https://gist.githubusercontent.com/mihaipopescu/756670a3de23ac714b3bd71dcc04e2e7/raw/14b8701a6599bf2ec30a54c83cf7566c59cf43ba/60023-UNI-18437.cs.log)
[60023-UNI-18437.cs.unc](https://gist.githubusercontent.com/mihaipopescu/756670a3de23ac714b3bd71dcc04e2e7/raw/4d4059766e9ee8cfc23db7784ef8ee2b06e4cd95/60023-UNI-18437.cs.unc)
[Uncrustify.CSharp.cfg](https://gist.githubusercontent.com/mihaipopescu/756670a3de23ac714b3bd71dcc04e2e7/raw/1d064c1fabb76ce328dd015ed40023207de20ca1/Uncrustify.CSharp.cfg)
[UNI-18437.cs](https://gist.githubusercontent.com/mihaipopescu/756670a3de23ac714b3bd71dcc04e2e7/raw/63c64c10211459aa06e047e6d5d5cf70b8ae808f/UNI-18437.cs)

Disclaimer:
---
Note: This is a generated issue from our internal issue tracker.
Expect a dry description which could be insufficient to properly understand the issue.
On the other hand, the support files should be sufficient to easily reproduce and debug the issue,
but please comment if it's still unclear and I will get back with clarifications.
See main discussion thread here #1223.

**Comment by mihaipopescu** [comment]
@CDanU Just to make sure, for indentation rules, if we add a new one that say fixes this issue, does that rule going to also fix bad code (what is generated today) to the expected ? Is this a good case for rerun configuration.

**Comment by CDanU** [comment]
That depends on how the new option will be implemented. I assume that most likely this will be the case.

I won't have time for the next few months to do anything significant though.

**Comment by gmaurel** [comment]
Fixed at 3484983ebdba4b13668f6737ab3644383a8c76de, on Feb 11 18:46:00 2018
